### PR TITLE
tyr: current jobs(state='running') with data_sets should not be deleted

### PR DESCRIPTION
**Pull request by kadhikari**
Please see: https://jira.kisio.org/browse/NAVP-973 for details

### CODE REVIEW

**Review by mbounabi — APPROVED** [review]

**Review by kinnou02 — COMMENTED** [review]
The purge run everyday use the `last_datasets()` method: https://github.com/CanalTP/navitia/blob/dev/source/tyr/tyr/tasks.py#L344

**Review by kinnou02 — CHANGES_REQUESTED** [review]
Since this method isn't used it won't solve the problem

File: source/navitiacommon/navitiacommon/models/__init__.py [hunk 1]
```diff
@@ -460,6 +460,20 @@ def last_datasets(self, nb_dataset=1, family_type=None):
...
+    def running_datasets(self):
+        """
+        return all datasets with job state = 'running' for this instance
+        """
+        data_sets = (
+            db.session.query(DataSet)
+            .join(Job)
+            .join(Instance)
+            .filter(Instance.id == self.id, Job.state == 'running')
+            .order_by(Job.created_at.desc())
```
Comment (pbougue) [inline source/navitiacommon/navitiacommon/models/__init__.py:472]: Why do you need that order?
If it's only for logging we can probably save it, or order it ascending as the more interesting (problematic) are the oldest ones.

File: source/tyr/tests/integration/purge_test.py [hunk 2]
```diff
@@ -241,3 +257,65 @@ def test_purge_old_jobs_of_multi_instance():
     assert len(jobs_resp['jobs']) == 1
     jobs_resp = api_get('/v0/jobs/fr')
     assert len(jobs_resp['jobs']) == 1
+
+
+@pytest.mark.usefixtures("init_instances_dir")
+def test_purge_old_jobs_with_diff_job_states():
+    """
+    we shouldn't delete jobs with status 'running'. This is to avoid
+    deletinng jobs and theirs files on running state.
```
Comment (pbougue) [inline source/tyr/tests/integration/purge_test.py]: typo: deleting (one `n`) jobs and their (no `s`) `while` running.

File: source/tyr/tyr/tasks.py [hunk 3]
```diff
@@ -344,7 +344,10 @@ def purge_instance(instance_id, nb_to_keep):
         os.path.realpath(os.path.dirname(dataset.name)) for dataset in instance.last_datasets(nb_to_keep)
     )
     logger.info('loaded  data are: %s', loaded)
-    to_remove = [os.path.join(instance_config.backup_directory, f) for f in backups - loaded]
+
+    running = set(os.path.realpath(os.path.dirname(dataset.name)) for dataset in instance.running_datasets())
+    logger.info('running  data are: %s', running)
```
Comment (pbougue) [inline source/tyr/tyr/tasks.py]: running `bina` are

**Review by pbougue — APPROVED** [review]

**Review by kinnou02 — APPROVED** [review]

**Pull request merged**